endpoints/v2/blob.py

```diff
@@ -332,6 +343,18 @@ def upload_chunk(namespace_name, repo_name, upload_uuid):
     # Upload the chunk for the blob.
     _upload_chunk(uploader)
 
+    if app.config.get("FEATURE_QUOTA_MANAGEMENT", False):
+        quota = quota_registry_model.verify_namespace_quota_during_upload(
+            namespace_name, repository_ref
+        )
+        if quota["severity_level"] == 2:
+            return Response(
+                status=413,
+                headers={
+                    "Range": _render_range(0),
+                },
+            )
```

Comment:
Shouldn't this check happen before anything else?
Looks like this check is needed in multiple places. You could wrap it in a decorator and apply it to the relevant endpoint. Similar to say, `@require_repo_write`